lib/vagrant/action/builtin/mixin_synced_folders.rb

```diff
@@ -190,6 +188,12 @@ def synced_folders(machine, **opts)
             folders[impl_name] = new_fs
           end
 
+          folders.each do |impl_name, fs|
+            fs.each do |id, data|
+              data[:impl] = plugins[impl_name][0].new()
```

Comment:
hmmmmm, it would be neat to have the synced folder object attached to the synced folder instead of just a hash of information. Would also cut down of the number of times the synced folders classes get instantiated. It seems like a new object is instantiated for about every action with the current implementation. 
I'm not completely convinced by this approach though. I don't like that each folder get's its own `:impl`. It would be better if each folder type got it's own `:impl` that is shared across folders of the same type.